pkg/chunk/cache/redis_client.go

```diff
@@ -0,0 +1,134 @@
+package cache
+
+import (
+	"context"
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"strings"
+	"time"
+	"unsafe"
+
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+
+	"github.com/go-redis/redis/v8"
+)
+
+// RedisConfig defines how a RedisCache should be constructed.
+type RedisConfig struct {
+	Endpoint    string         `yaml:"endpoint"`
+	MasterName  string         `yaml:"master_name"`
+	Timeout     time.Duration  `yaml:"timeout"`
+	Expiration  time.Duration  `yaml:"expiration"`
+	PoolSize    int            `yaml:"pool_size"`
+	Password    flagext.Secret `yaml:"password"`
+	EnableTLS   bool           `yaml:"enable_tls"`
+	IdleTimeout time.Duration  `yaml:"idle_timeout"`
+	MaxConnAge  time.Duration  `yaml:"max_connection_age"`
+}
+
+// RegisterFlagsWithPrefix adds the flags required to config this to the given FlagSet
+func (cfg *RedisConfig) RegisterFlagsWithPrefix(prefix, description string, f *flag.FlagSet) {
+	f.StringVar(&cfg.Endpoint, prefix+"redis.endpoint", "", description+"Redis Server endpoint to use for caching. A comma-separated list of endpoints for Redis Cluster or Redis Sentinel. If empty, no redis will be used.")
+	f.StringVar(&cfg.MasterName, prefix+"redis.master_name", "", description+"Redis Sentinel master name. An empty string for Redis Server or Redis Cluster.")
```

Comment:
Sorry for nitpicking, but we try to keep YAML and CLI names consistent, except in YAML we use underscore and in CLI options we use dashes, so this should really be `redis.master-name`.